sound/soc/bcm/hifiberry_digi.c

```diff
@@ -163,33 +161,47 @@ static struct snd_soc_dai_link snd_rpi_hifiberry_digi_dai[] = {
 
 /* audio machine driver */
 static struct snd_soc_card snd_rpi_hifiberry_digi = {
-	.name         = "snd_rpi_hifiberry_digi",
 	.owner        = THIS_MODULE,
 	.dai_link     = snd_rpi_hifiberry_digi_dai,
 	.num_links    = ARRAY_SIZE(snd_rpi_hifiberry_digi_dai),
 };
 
 static int snd_rpi_hifiberry_digi_probe(struct platform_device *pdev)
 {
+	struct snd_soc_card *card = &snd_rpi_hifiberry_digi;
+	struct device_node *np = pdev->dev.of_node;
+	char *prefix = "wm8804-digi,";
+	char prop[128];
 	int ret = 0;
 
 	snd_rpi_hifiberry_digi.dev = &pdev->dev;
 
-	if (pdev->dev.of_node) {
+	if (np) {
 	    struct device_node *i2s_node;
 	    struct snd_soc_dai_link *dai = &snd_rpi_hifiberry_digi_dai[0];
-	    i2s_node = of_parse_phandle(pdev->dev.of_node,
-					"i2s-controller", 0);
+	    i2s_node = of_parse_phandle(np, "i2s-controller", 0);
 
 	    if (i2s_node) {
 		dai->cpu_dai_name = NULL;
 		dai->cpu_of_node = i2s_node;
 		dai->platform_name = NULL;
 		dai->platform_of_node = i2s_node;
 	    }
+
+	    snprintf(prop, sizeof(prop), "%scard-name", prefix);
+	    if (of_property_read_string(np, prop, &card->name))
+		card->name = "snd_rpi_hifiberry_digi";
```

Comment:
The comments for of_property_read_string say:
```
 * @out_string: pointer to null terminated return string, modified only if
 *              return value is 0.
```
If you leave the default strings in snd_rpi_hifiberry_digi & _dai then you can replace these conditionals with simple calls to of_property_read_string and get the same result.